src/plugins/saved_objects_management/server/routes/find.test.ts

```diff
@@ -0,0 +1,158 @@
+  let router: jest.Mocked<IRouter>;
+  let managementService: jest.Mocked<ISavedObjectsManagement>;
+  let context: any;
+  let req: any;
+  let res: any;
+
+  beforeAll(() => {
+    router = httpServiceMock.createRouter();
+
+    // Mock management service
+    managementService = managementMock.create();
+
+    // Mock context with savedObjects client
+    const savedObjectsClient = {
+      find: jest.fn(),
+      get: jest.fn(),
+    };
+
+    context = {
+      core: {
+        savedObjects: {
+          client: savedObjectsClient,
+        },
+      },
+    };
+
+    // Mock request
+    req = {
+      query: {
+        perPage: 20,
+        page: 1,
+        type: 'test-type',
+        fields: ['field1', 'field2'],
+      },
+    };
+
+    // Mock response
+    res = {
+      ok: jest.fn(),
+    };
+  });
+
+  it('should register GET route with correct path', () => {
+    registerFindRoute(router, Promise.resolve(managementService));
+
+    expect(router.get).toHaveBeenCalledWith(
+      expect.objectContaining({
+        path: '/api/opensearch-dashboards/management/saved_objects/_find',
+      }),
+      expect.any(Function)
+    );
+  });
+
+  it('should handle basic find request', async () => {
+    registerFindRoute(router, Promise.resolve(managementService));
+
+    const findResponse = {
+      saved_objects: [
+        {
+          type: 'test-type',
+          id: '1',
+          attributes: {
+            field1: 'value1',
+            field2: 'value2',
+          },
+        },
+      ],
+      total: 1,
+      per_page: 20,
+      page: 1,
+    };
+
+    context.core.savedObjects.client.find.mockResolvedValue(findResponse);
+
+    const handler = router.get.mock.calls[0][1];
+    await handler(context, req, res);
+
+    expect(context.core.savedObjects.client.find).toHaveBeenCalledWith(
+      expect.objectContaining({
+        perPage: 20,
+        page: 1,
+        type: 'test-type',
+      })
+    );
+    expect(res.ok).toHaveBeenCalled();
+  });
+
+  it('should handle index-pattern type correctly', async () => {
+    registerFindRoute(router, Promise.resolve(managementService));
+
+    const findResponse = {
+      saved_objects: [
+        {
+          type: 'index-pattern',
+          id: '1',
+          attributes: {
+            title: 'test-pattern',
+          },
+          references: [],
+        },
+      ],
+      total: 1,
+      per_page: 20,
+      page: 1,
+    };
+
+    context.core.savedObjects.client.find.mockResolvedValue(findResponse);
+    context.core.savedObjects.client.get.mockResolvedValue({ attributes: {} });
+
+    const handler = router.get.mock.calls[0][1];
+    await handler(context, req, res);
+
+    expect(context.core.savedObjects.client.find).toHaveBeenCalled();
+    expect(res.ok).toHaveBeenCalled();
+  });
+
+  it('should handle workspace filtering', async () => {
+    registerFindRoute(router, Promise.resolve(managementService));
+
+    req.query.workspaces = ['workspace1'];
```

Comment:
nit: should we have a case for workspace is empty?

Reply:
I think the no workspace case is already been covered by the above "basic find" test case. For a find request without workspace filtering then the saved object is always opt-in so that it somehow pointless to explicitly test this case.